lib/shared/robot/README.md

```diff
@@ -0,0 +1,89 @@
+The "robot" mechanism
+=====================
+
+As discussed in `lib/child-writer`, communication between the main Etcher
+application and its elevated writer process happens through an IPC (Inter
+Process Communication) channel. In a nutshell, we emit every single line that
+the writer process prints to the parent as a "message". Since these "lines"
+need to convey non-trivial information such as progress information, speed,
+final computer checksums, etc we are in need of a basic form of a "protocol".
+
+The "robot" module is the entity that implements this protocol, and provides
+utility functions to read/send messages using the protocol targetted at both
+parties (the client and the writer processes).
+
+The codename "robot" is inspired by [xz][xz-man], which provides a `--robot`
+option that makes the tool print machine-parseable output:
+
+```
+--robot
+    Print messages in	a machine-parsable format.  This  is  intended
+    to  ease	writing	 frontends  that  want	to  use	 xz instead of
+    liblzma, which may be the	case with various scripts.  The	output
+    with  this  option  enabled  is  meant  to  be  stable across xz
+    releases.	 See the section ROBOT MODE for	details.
+```
+
+To enable the "robot" option, we standardised the presence of an
+`ETCHER_CLI_ROBOT` environment variable. You can check if the mode is enabled
+by using the `.isEnabled()` function.
```

Comment:
Where is the method `.isEnabled()` called from? Is it the instance of the `robot` module, and if so should this be clarified here?

I know it says on line 11 about the robot module, and so from context it should be relatively easy to tell what it's referring to. I'm just wondering whether it would be better to have it here also or maybe make it read `robot.isEnabled()` function (or perhaps present a code sample).